File: app/src/main/java/com/mikepenz/fastadapter/app/adapters/MopubFastItemAdapter.java
package com.mikepenz.fastadapter.app.adapters;

import com.mikepenz.fastadapter.IItem;
import com.mikepenz.fastadapter.commons.adapters.FastItemAdapter;
import com.mopub.nativeads.MoPubRecyclerAdapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by mikepenz on 28.06.16.
 */

public class MopubFastItemAdapter<Item extends IItem<? extends RecyclerView.ViewHolder>> extends FastItemAdapter<Item> {
    private MoPubRecyclerAdapter mMoPubAdAdapter;

    public MopubFastItemAdapter() {
        setLegacyBindViewMode(true);
    }

    public MopubFastItemAdapter withMoPubAdAdapter(MoPubRecyclerAdapter moPubAdAdapter) {
        this.mMoPubAdAdapter = moPubAdAdapter;
        return this;
    }

    @Override
    public int getHolderAdapterPosition(@NonNull RecyclerView.ViewHolder holder) {
        return mMoPubAdAdapter.getOriginalPosition(super.getHolderAdapterPosition(holder));
    }
}

File: app/src/main/java/com/mikepenz/fastadapter/app/adapters/MopubFastItemAdapter.kt
package com.mikepenz.fastadapter.app.adapters

import androidx.recyclerview.widget.RecyclerView
import com.mikepenz.fastadapter.GenericItem
import com.mikepenz.fastadapter.adapters.FastItemAdapter
import com.mopub.nativeads.MoPubRecyclerAdapter

/**
 * Created by mikepenz on 28.06.16.
 */

class MopubFastItemAdapter<Item : GenericItem> : FastItemAdapter<Item>() {
    private var mMoPubAdAdapter: MoPubRecyclerAdapter? = null

    init {
        legacyBindViewMode = true
    }

    fun withMoPubAdAdapter(moPubAdAdapter: MoPubRecyclerAdapter): MopubFastItemAdapter<*> {
        this.mMoPubAdAdapter = moPubAdAdapter
        return this
    }

    override fun getHolderAdapterPosition(holder: RecyclerView.ViewHolder): Int {
        return mMoPubAdAdapter!!.getOriginalPosition(super.getHolderAdapterPosition(holder))
    }
}
